docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,111 @@
+
+1. Start with [scripts as code](#scripts-as-code) - this method offers the best cost/reward ratio, is the simplest to implement and maintain over time.
+2. Move towards [project as code](#project-as-code) - this method is more difficult to set up, but comes with the benefit of having your deployment managed as code.
+
+!toc
+
+## Scripts as code {#scripts-as-code}
+
+The simplest way to get started with **deployment as code** is to manage your custom deployment **scripts as code**. When you deploy your application, Octopus can execute a script contained inside a package. You can colocate your deployment scripts with your application source code, leveraging all the benefits of source control including change tracking and branching, then package it all up for Octopus.
+
+There is a downside to stopping here: your scripts are managed as code, but your deployment process and variables are still controlled by Octopus. Depending on your situation, this tradeoff might be quite acceptable.
+
+### Move your scripts into source code
+
+You can follow this process to move your custom scripts without interrupting deployments:
+
+1. Move your deployment scripts from your Octopus deployment process into a file in your application source code.
+1. Build your application as normal, but this time with the deployment scripts packaged up alongside your application.
+  a. At this point your deployments will continue to work using the scripts stored in Octopus.
+1. Update your deployment process to use the script from your application package.
+
+Now your scripts are colocated with your application source code, all without changing your build pipeline.
+
+Learn about [executing custom scripts in packages](/docs/deployment-examples/custom-scripts/scripts-in-packages/index.md).
+Learn about [custom scripts](/docs/deployment-examples/custom-scripts/index.md).
+
+### Consistency and repeatability using scripts as code
+
+When you manage your **scripts as code**, Octopus still makes sure your deployments are consistent and repeatable. Whenever you modify a script, that change flows through the whole process just like the changes to your application code.
+
+### Introducing changes safely using scripts as code {#scripts-as-code-change-safely}
+
+Branches in source control let you test application code changes in isolation before integrating them with other code changes. You can use the exact same approach to introduce changes to your scripts as code. This lets you make changes to your scripts without breaking deployments from your `master` branch.
+
+- **Modifying an existing script**: If you modify a script in a branch, that change flows through the whole process just like the changes to your application code. When you deploy a release from that branch, your modified script will be used. When you merge your branch into the `master` branch, your modified script will be used for deployments from the `master` branch.
```

Comment:
Does this make an assumption that people are using `master` for production, and branches for feature/dev work? 
Worth calling that out?
Also, does have an implicit assumption that people are using git?